server/game/cards/06-WoE/EndlessHordes.js

```diff
@@ -0,0 +1,67 @@
+const Card = require('../../Card.js');
+
+class EndlessHordes extends Card {
+    // Play: For each enemy creature, make a token creature, ready it,
+    // and have it fight that enemy creature, ignoring taunt. Resolve
+    // these fights one at a time.
+    //
+    // Note the FAQ entry states that all the tokens are created first,
+    // then all readied, then fight targets are chosen one at a time and
+    // the fights are fully resolved.
+    //
+    // Unofficial ruling from blinkingline is that only enemy
+    // creatures that were present at the beginning of resolution can
+    // be targeted for later fighting.  If multiple creatures are
+    // killed during one fight, then the leftover tokens just end up
+    // ready (even if new creatures were created during earlier
+    // fights).
+    setupCardAbilities(ability) {
+        this.validTargets = [];
+        this.newTokens = [];
+
+        this.play({
+            gameAction: ability.actions.makeTokenCreature((context) => {
+                this.validTargets = context.player.opponent.creaturesInPlay.slice();
+                return {
+                    amount: this.validTargets.length
+                };
+            }),
+            then: {
+                gameAction: ability.actions.sequentialForEach((context) => {
```

Comment:
Here you can avoid the `sequentialForEach` using `ready`action with target = `context.preThenEvent.cards`